zerver/lib/send_email.py

```diff
@@ -147,23 +137,25 @@ def send_future_email(template_prefix: str, realm: Realm, to_user_ids: Optional[
         # For logging the email
 
     assert (to_user_ids is None) ^ (to_emails is None)
-    if to_user_ids is not None:
-        # The realm is redundant if we have a to_user_id; this assert just
-        # expresses that fact
-        assert(len(to_user_ids) == 1)
-        assert(UserProfile.objects.filter(id__in=to_user_ids, realm=realm).exists())
-        to_field = {'user_id': to_user_ids[0]}  # type: Dict[str, Any]
-    else:
-        assert to_emails is not None
-        assert(len(to_emails) == 1)
-        to_field = {'address': parseaddr(to_emails[0])[1]}
-
-    ScheduledEmail.objects.create(
+    email = ScheduledEmail.objects.create(
         type=EMAIL_TYPES[template_name],
         scheduled_timestamp=timezone_now() + delay,
         realm=realm,
-        data=ujson.dumps(email_fields),
-        **to_field)
+        data=ujson.dumps(email_fields))
```

Comment:
Can we structure this by using `ScheduledEmail(` instead of the `.create()` method, and just have one `.save()` call at the end, to eliminate the need to have the except/delete block at the bottom?

Reply:
As per https://docs.djangoproject.com/en/2.1/topics/db/examples/many_to_many/. We can't associate users with scheduled emails until it's been saved.

Reply:
OK, makes sense.  I guess we need to go with this approach then.